src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/dates/ReleaseDateService.kt

```diff
@@ -146,21 +130,14 @@ class ReleaseDateService(
     nomisRecord.determineLicenceStartDate()
   }
 
-  private fun calculateHardStopDate(conditionalReleaseDate: LocalDate): LocalDate {
-    val date =
-      if (conditionalReleaseDate.isNonWorkingDay()) 1.workingDaysBefore(conditionalReleaseDate) else conditionalReleaseDate
-    return 2.workingDaysBefore(date)
-  }
-
-  private fun isExcludedFromHardstop(actualReleaseDate: LocalDate, latestReleaseDate: LocalDate): Boolean {
-    if (latestReleaseDate.isNonWorkingDay()) {
-      return actualReleaseDate != 1.workingDaysBefore(latestReleaseDate)
-    }
-    return actualReleaseDate != latestReleaseDate
+  private fun calculateHardStopDate(licenceStartDate: LocalDate): LocalDate {
+    val adjustedLsd =
+      if (licenceStartDate.isNonWorkingDay()) 1.workingDaysBefore(licenceStartDate) else licenceStartDate
+    return 2.workingDaysBefore(adjustedLsd)
   }
 
   fun getHardStopWarningDate(sentenceDateHolder: SentenceDateHolder): LocalDate? {
-    val hardStopDate = getHardStopDate(sentenceDateHolder) ?: return null
+    val hardStopDate = getHardStopDate(sentenceDateHolder.licenceStartDate) ?: return null
```

Comment:
Seems like there are now a few places where `sentenceDateHolder` is being passed in just to access the licence start date.
Perhaps we can do a tidy-up ticket following this one to simplify, but I suspect it may spiral into touching quite a few files, so probably best not to do it in this PR/ticket.

Reply:
sounds a good idea :-)